index.js

```diff
@@ -7,65 +7,48 @@ var through = require('through2');
+ * @param {Object=} opt_clangFormat A clang-format module to optionally use.
+ */
+function checkFormat(opt_options, opt_clangFormat) {
+  var optsStr = opt_options || {BasedOnStyle: 'Google'};
+  var actualClangFormat = opt_clangFormat || clangFormat;
+  if (typeof optsStr === 'object') {
+    optsStr = JSON.stringify(optsStr);
+  }
 
-    return through.obj(
-        function(file, enc, done) {
-          var actualStream =
-              file.isStream() ? file.content :
-                                fs.createReadStream(file.path, {encoding: enc});
-          var expectedStream = clangFormat(file, enc, optsStr, done);
-          streamEqual(actualStream, expectedStream, function(err, equal) {
-            if (err) {
-              return done(err);
-            }
-            if (!equal) {
-              errors.push(file.path);
-            }
-            done();
-          });
-        },
-        function(cb) {
-          if (errors.length > 0) {
-            // Avoid version skew between the version installed under
-            // gulp-clang-format and any
-            // version that was installed globally
-            var clangFormatBin =
-                './node_modules/gulp-clang-format/node_modules/clang-format/index.js';
-            gutil.log('WARNING: Files are not properly formatted. Please run');
-            gutil.log('  ' + clangFormatBin + ' -i -style="' + optsStr + '" ' +
-                      errors.join(' '));
-            gutil.log('  (using clang-format version ' + getClangFormatVersion() + ')');
-            this.emit('warning', new gutil.PluginError('gulp-clang-format',
-                                                       'files not formatted'));
-          }
-          cb();
-        });
+  function filter(file, enc, done) {
+    var actualStream =
+        file.isStream() ? file.content : fs.createReadStream(file.path, {encoding: enc});
+    var expectedStream = actualClangFormat(file, enc, optsStr, done);
+    streamEqual(actualStream, expectedStream, function(err, equal) {
+      if (err) {
+        return done(err);
+      }
+      if (!equal) {
+        errors.push(file.path);
+      }
+      done();
+    });
   }
-};
 
-function getClangFormatVersion() {
-  var pkg;
-  var cf = require.resolve('clang-format');
-  var cfPath = cf;
-  while ((cfPath = path.dirname(cfPath)) !== '/') {
-    pkg = path.join(cfPath, 'package.json');
-    if (fs.existsSync(pkg)) {
-      var content = JSON.parse(fs.readFileSync(pkg, 'utf-8'));
-      return content['version'];
+  function onError(cb) {
+    if (errors.length > 0) {
+      var clangFormatBin = './' + path.relative(process.cwd(), clangFormat.location);
+      gutil.log('WARNING: Files are not properly formatted. Please run');
+      gutil.log('  ' + clangFormatBin + ' -i -style="' + optsStr + '" ' + errors.join(' '));
+      gutil.log('  (using clang-format version ' + clangFormat.version + ')');
```

Comment:
I wonder when someone sees this message, will they know whether to blame the version or not? I guess they just need to ensure it's the latest published one?